src/Tribe/Query.php

```diff
@@ -1151,14 +1150,37 @@ public static function getEvents( $args = array(), $full = false ) {
 			$cache_key = 'get_events_' . get_current_user_id() . serialize( $args );
 
 			$result = $cache->get( $cache_key, 'save_post' );
-			if ( $result && $result instanceof WP_Query ) {
+
+			if (
+				false !== $result
+				&& (
+					$result instanceof WP_Query
+					|| (
+						$return_found_posts
+						&& is_int( $result )
+					)
+				)
+			) {
 				do_action( 'log', 'cache hit', 'tribe-events-cache', $args );
 			} else {
 				do_action( 'log', 'no cache hit', 'tribe-events-cache', $args );
-				$result = new WP_Query( $args );
+
+				/** @var Tribe__Events__Repositories__Event $event_orm */
+				$event_orm = tribe( 'events.event-repository' );
```

Comment:
Why not `tribe_events` here? In so doing all the filtering and decoration that is applied on top is lost.
Is this intentional to specifically use only the version of the repository supplied by TEC and PRO?
E.g. tickets decorator would be left out here.

Reply:
✅